### ISSUE TITLE
Keras Inception import error: Paramter gamma does not exist in weights

### ISSUE THREAD

**Issue by ClnSchlssr** [deeplearning4j/deeplearning4j#4848]
#### Issue Description

I'm trying to import an inception v3 model made in keras using 

`KerasModelImport.importKerasModelAndWeights(modelFile.getAbsolutePath(),false)
`
I get the error: InvalidKerasConfigurationException: Parameter gamma does not exist in weights.
The exception gets thrown when trying to set the gamma parameter which doesn't seem to get loaded properly (unlike beta, moving_mean and moving_variance).

Here is the model in case anything stands out:

```python
def inception(dim, categories, dense_size = 512, dropout = 0.5):
    base_model = applications.inception_v3.InceptionV3(
            weights="imagenet", include_top=False, input_shape=(dim,dim,3))
    x = base_model.output
    x = GlobalAveragePooling2D()(x)
    x = Dense(dense_size, activation='relu')(x)
    x = Dropout(dropout)(x)
    if categories == 2:
        probabilities = Dense(1)(x)
        predictions = Activation('sigmoid')(probabilities)
    else:
        probabilities = Dense(categories)(x)
        predictions = Activation('softmax')(probabilities)
    model = Model(inputs=base_model.input, outputs=predictions)   
return (model, base_model)
```

#### Version Information

DL4J ver: 1.9.2 - SNAPSHOT
keras ver: 2.1.3
OS: Windows 7

My instincts with these libraries are not great. Where should I direct my attention to track this down. Any clues would be appreciated.

**Comment by maxpumperla** [comment]
@ClnSchlssr can you provide me with an h5 file directly?

**Comment by ClnSchlssr** [comment]
Sent to the email listed on your profile

**Comment by tomthetrainer** [comment]
I was able to import inceptionv3 into dl4j with the following combination of keras and dl4j. 

Note this was a while ago, Keras 1. 

I used a theano backend. 

This might not help  @maxpumperla in patching dl4j, but it might help @ClnSchlssr to get a working version. 

This should be a dl4j version of inceptionv3 saved using modelserializer. 

See the files here. 
https://github.com/SkymindIO/Demos-for-sales-marketting/releases

If my sloppy archive is accurate I generated the keras h5 files using this code. 

https://gist.github.com/tomthetrainer/9059546f30bf3d944c8826170e6d5009

Basically some slight modification of fchollet's examples here

https://github.com/fchollet/deep-learning-models

The Java to load it is here

https://gist.github.com/tomthetrainer/c6d994cfeca34378598c3d699f38af0b

**Comment by maxpumperla** [comment]
thanks @ClnSchlssr, I'll look into this soon

**Comment by mirismr** [comment]
How can I get that version which fix that issue?

**Comment by AlexDBlack** [comment]
@mirismr This is available in the recently release 1.0.0-beta version of DL4J, which is on maven central with all the other releases.

**Comment by mirismr** [comment]
This is my pom.xml file, but I'm still getting an error... any ideas?
https://gist.github.com/mirismr/ae0e706d52adb9dbee84b7edf0aab4a9

**Comment by ClnSchlssr** [comment]
@maxpumperla , I think this issue has surfaced again because if a change to KerasBatchNormalization.java

Your fix created a dummy array of ones using beta to get the dimensions:

```
        if (scale) {
            if (weights.containsKey(PARAM_NAME_GAMMA))
                this.weights.put(BatchNormalizationParamInitializer.GAMMA, weights.get(PARAM_NAME_GAMMA));
            else
                throw new InvalidKerasConfigurationException(
                        "Parameter " + PARAM_NAME_GAMMA + " does not exist in weights");
        } else {
            INDArray dummyGamma = Nd4j.onesLike(weights.get(PARAM_NAME_BETA));
            this.weights.put(BatchNormalizationParamInitializer.GAMMA, dummyGamma);

```
But in the current implementation someone has changed beta->gamma which by the programs own logic shouldn't be available and I am indeed getting a null pointer exception when I try to import models. 


```
if (scale) {
            if (weights.containsKey(PARAM_NAME_GAMMA))
                this.weights.put(BatchNormalizationParamInitializer.GAMMA, weights.get(PARAM_NAME_GAMMA));
            else
                throw new InvalidKerasConfigurationException(
                        "Parameter " + PARAM_NAME_GAMMA + " does not exist in weights");
        } else {
            INDArray dummyGamma = Nd4j.onesLike(weights.get(PARAM_NAME_GAMMA));
            this.weights.put(BatchNormalizationParamInitializer.GAMMA, dummyGamma);


```

Let me know if I'm missing anything here. Otherwise it can just be changed back to beta?

Best,
Colin

**Comment by maxpumperla** [comment]
sorry. should be an easy fix

**Comment by ClnSchlssr** [comment]
No problem. Thanks for looking into it.

**Comment by asvk** [comment]
I have the same exception with inception model. 
After code review `https://github.com/keras-team/keras/blob/master/keras/layers/normalization.py#L61-L72` looks like if we don't have "scale" parameter, we don't have "gamma" too. 

I tried to merge your solutions in one pull request. 
`
https://github.com/deeplearning4j/deeplearning4j/pull/5425
`
At least it helped in my case.

**Comment by maxpumperla** [comment]
@asvk thanks for your PR!
